### PR TITLE
Add file with nesting two-level list structure

### PR DESCRIPTION
**Pull request by wgtmac**
See: https://github.com/apache/arrow/issues/43994

The Parquet file was created by parquet-java 1.14.3 with following code:
```java
package org.example;

import org.apache.avro.Schema;
import org.apache.avro.generic.GenericData;
import org.apache.avro.generic.GenericRecord;
import org.apache.hadoop.conf.Configuration;
import org.apache.hadoop.fs.Path;
import org.apache.parquet.avro.AvroParquetWriter;
import org.apache.parquet.hadoop.ParquetWriter;

import java.io.IOException;
import java.util.Arrays;
import java.util.stream.Collectors;
import java.util.stream.Stream;

public class TwoLevelList {

  public static void main(String[] args) {
    Schema schema = new Schema.Parser().parse("{"
      + "\"type\":\"record\","
      + "\"name\":\"my_record\","
      + "\"fields\":["
      + "  {"
      + "    \"name\":\"a\","
      + "    \"type\":{\"type\":\"array\", \"items\":{\"type\":\"array\", \"items\":\"int\"}}"
      + "  }"
      + "]"
      + "}");

    GenericRecord record = new GenericData.Record(schema);

    // Write [[1, 2], [3, 4]] to the avro record
    record.put("a", Stream.of(Arrays.asList(1, 2), Arrays.asList(3, 4))
      .map(list -> {
        Schema innerListType = schema.getField("a").schema().getElementType();
        GenericData.Array<Integer> innerList = new GenericData.Array<>(list.size(), innerListType);
        innerList.addAll(list);
        return innerList;
      }).collect(Collectors.toList()));

    Path file = new Path("/tmp/old_list_structure.parquet");
    Configuration conf = new Configuration();
    conf.set("parquet.avro.write-old-list-structure", "true");  // this is the default value
    try (ParquetWriter<GenericRecord> writer = AvroParquetWriter.<GenericRecord>builder(file)
      .withSchema(schema)
      .withConf(conf)
      .build()) {
      writer.write(record);
    } catch (IOException e) {
      throw new RuntimeException(e);
    }
  }

}

```

File metadata using parquet-cli:
```
> parquet-cli meta /tmp/old_list_structure.parquet

File path:  /tmp/test.parquet
Created by: parquet-mr version 1.14.3 (build b5e376a2caee767a11e75b783512b14cf8ca90ec)
Properties:
  parquet.avro.schema: {"type":"record","name":"my_record","fields":[{"name":"a","type":{"type":"array","items":{"type":"array","items":"int"}}}]}
    writer.model.name: avro
Schema:
message my_record {
  required group a (LIST) {
    repeated group array (LIST) {
      repeated int32 array;
    }
  }
}


Row group 0:  count: 1  53.00 B records  start: 4  total(compressed): 53 B total(uncompressed):53 B
--------------------------------------------------------------------------------
               type      encodings count     avg size   nulls   min / max
a.array.array  INT32     _   _     4         13.25 B    0       "1" / "4"
```

Data using parquet-cli:
```
> parquet-cli cat /tmp/old_list_structure.parquet

{"a": [[1, 2], [3, 4]]}
```

### CODE REVIEW

**Comment by wgtmac** [comment]
@mapleFU @pitrou Could you please take a look? Thanks!

**Comment by pitrou** [comment]
I don't understand how the structure here is different from the "list of lists" example in https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists

Witness:
```
required group a (LIST) {
  repeated group list {
    required group element (LIST) {
      repeated group list {
        required int32 element;
      }
    }
  }
}
```

vs.
```
optional group array_of_arrays (LIST) {
  repeated group list {
    required group element (LIST) {
      repeated group list {
        required int32 element;
      }
    }
  }
}
```

**Comment by wgtmac** [comment]
My bad! I pasted the wrong result which disabled `parquet.avro.write-old-list-structure`. Now the description is fixed. @pitrou

File: data/README.md [hunk 1]
```diff
@@ -53,6 +53,7 @@
 | incorrect_map_schema.parquet | Contains a Map schema without explicitly required keys, produced by Presto. See [note](#incorrect-map-schema) |
 | column_chunk_key_value_metadata.parquet | two INT32 columns, one with column chunk key-value metadata {"foo": "bar", "thisiskeywithoutvalue": null} note that the second key "thisiskeywithoutvalue", does not have a value, but the value can be mapped to an empty string "" when read depending on the client |
 | sorting_columns.parquet | INT64 and BYTE_ARRAY columns with first column with nulls first and descending, second column with nulls last and ascending. This file contains two row groups with same data and sorting columns. |
+| old_list_structure.parquet | Single LIST<LIST<INT32>> column with legacy two-level list structure |
```
Comment (pitrou) [inline data/README.md:56]: Can you document the file like you did in the PR description?

Comment (wgtmac) [reply data/README.md]: Added. Please take a look again. Thanks!

File: data/old_list_structure.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      throw new RuntimeException(e);
+    }
+  }
+
+}
+```
+
+Here is the file metadata printed by parquet-cli:
+```
+File path:  /tmp/old_list_structure.parquet
```
Comment (mapleFU) [inline data/old_list_structure.md:84]: should /tmp be here or not?

File: data/old_list_structure.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+Created by: parquet-mr version 1.14.3 (build b5e376a2caee767a11e75b783512b14cf8ca90ec)
```
Comment (mapleFU) [inline data/old_list_structure.md:85]: Is this build a valid release? Or just run by master?

**Review by mapleFU — APPROVED** [review]
General LGTM

File: data/old_list_structure.md (hunk 3 above)
Comment (wgtmac) [reply data/old_list_structure.md:85]: This is a released version.

**Comment by mapleFU** [comment]
Merged, thanks all!

**Pull request merged**